contracts/RevisedPixelmonTrainerAdventure/PxTrainerAdventure.sol

```diff
@@ -0,0 +1,380 @@
+                if (week.distributions[index].totalSupply == 0 || !week.winners[msg.sender].treasureTypeClaimed[treasureType]) {
+                    continue;
+                }
+                unchecked {
+                    sumOfTotalSupply += week.distributions[index].totalSupply;
+                }
+                if (random <= sumOfTotalSupply) {
+                    selectedIndex = index;
+                    break;
+                }
+            }
+        } else {
+            uint256 random = randomNumber - ((randomNumber / remaining) * remaining) + 1;
+
+            for (uint256 index = 1; index <= week.treasureCount; index = _uncheckedInc(index)) {
+                uint256 treasureType = treasures[week.distributions[index].treasureIndex].treasureType;
+                if (week.distributions[index].totalSupply == 0 || week.winners[msg.sender].treasureTypeClaimed[treasureType]) {
+                    continue;
+                }
+                unchecked {
+                    sumOfTotalSupply += week.distributions[index].totalSupply;
+                }
+                if (random <= sumOfTotalSupply) {
+                    selectedIndex = index;
+                    break;
+                }
+            }
+        }
+
+        uint256 selectedTreasureIndex = week.distributions[selectedIndex].treasureIndex;
+        week.winners[msg.sender].treasureTypeClaimed[treasures[selectedTreasureIndex].treasureType] = true;
+        unchecked {
+            week.distributions[selectedIndex].totalSupply--;
+            week.winners[msg.sender].claimed++;
+            week.remainingSupply--;
+            treasures[selectedTreasureIndex].claimedToken++;
+        }
+
+        transferToken(_weekNumber, treasures[selectedTreasureIndex]);
+    }
+
+    /// @notice Transfers token from vault to the method caller's wallet address
+    /// @dev This method will be used in a public method and user who call the
+    ///      method will get a token from vault wallet address
+    /// @param _treasure Treasure to transfer
+    function transferToken(uint256 _weekNumber, Treasure memory _treasure) internal {
+        if (_treasure.contractType == ERC_1155_TYPE) {
+            IERC1155 erc1155Contract = IERC1155(_treasure.collectionAddress);
+            erc1155Contract.safeTransferFrom(vaultWalletAddress, msg.sender, _treasure.tokenId, 1, "");
+        }
+        if (_treasure.contractType == ERC_721_TYPE) {
+            IERC721 erc721Contract = IERC721(_treasure.collectionAddress);
+            if (_treasure.tokenIds.length == _treasure.claimedToken) {
+                revert InsufficientToken();
+            }
+            erc721Contract.transferFrom(vaultWalletAddress, msg.sender, _treasure.tokenIds[_treasure.claimedToken - 1]);
+        }
+
+        emit TreasureTransferred(_weekNumber, msg.sender, _treasure.collectionAddress, _treasure.tokenId, _treasure.contractType);
+    }
+
+    /// @notice Set treasure treasure distributions for a week
```

Comment:
Mistype in the comment

_Set treasure treasure distributions for a week_

Should be

_Set treasure distributions for a week_

Reply:
fixed